File: c++/inheritance/src/dog.cpp
#include "dog.h"

dog::dog(string name, string owner) : animal(name){
    this->owner = owner;
}

string dog::get_owner(){
    return this->owner;
}

void dog::set_owner(string owner){
    this->owner = owner;
}

string dog::toString()const{
    return string(animal::toString()+string("\nowner:"+this->owner));
}

dog::~dog()
{
}

File: python/inheritance/dog.py
from animal import Animal

class Dog(Animal):
    __owner = ""

    def __init__(self, name, height, owner):
        super().__init__(name, height)
        self.__owner = owner
    def __str__(self):
        return super().__str__()+"owner:{}".format(self.__owner)
    
    def get_owner(self):
        return self.__owner
    def set_owner(self, owner):
        self.__owner = owner
    
    def get_type(self):
        print("Dog")

    def test(self):
        print(Dog._name)